### PR TITLE
Show number of features affected by in place editing

### LINKED ISSUE #38680: When running in place "repair geometries" with a selection it is unclear if it only affects the selected features
Run "repair geometries" in place with a selection. The algorithm is started without a dialog and reports "successfully finished". It is unclear to the user if it was selected only or not. Checking in the processing history, the flag "selectedOnly=True" can be seen.

I see two options:

 - Offer a dialog to let the user check/uncheck "selected only"
 - Report the number of features in the confirmation dialog of the in place editing

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes #38680

image

### CODE REVIEW

**Comment by m-kuhn** [comment]
@elpaso @nyalldawson  what's your opinion on the approach to return the number of features?

File: python/plugins/processing/gui/ProcessingToolbox.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def executeAlgorithm(self):
                 feedback = MessageBarProgress(algname=alg.displayName())
                 ok, results = execute_in_place(alg, parameters, feedback=feedback)
                 if ok:
-                    iface.messageBar().pushSuccess('', self.tr('{} complete').format(alg.displayName()))
+                    iface.messageBar().pushSuccess('', self.tr('{algname} completed. {featurecount} feature(s) processed.').format(algname=alg.displayName(), featurecount=results['count']))
```
Comment (nirvn) [inline python/plugins/processing/gui/ProcessingToolbox.py:256]: What about using a %n tr count here? i.e. self.tr('{algname} completed. {featurecount}').format(algname=...,featurecount=self.tr('%n feature(s) processed','',results['count'])) ?

Comment (m-kuhn) [reply python/plugins/processing/gui/ProcessingToolbox.py:256]: Good point, I was trying to use `tr().arg()` which doesn't work ... But that's not actually required!

File: python/plugins/processing/gui/AlgorithmExecutor.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ def execute_in_place_run(alg, parameters, context=None, feedback=None, raise_exc
 
                 feedback.setProgress(int((current + 1) * step))
 
-            results, ok = {}, True
+            results, ok = {'count': current + 1}, True
```
Comment (nyalldawson) [inline python/plugins/processing/gui/AlgorithmExecutor.py:210]: You'd also need to add this to the "#Tradition run..." block results below

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**